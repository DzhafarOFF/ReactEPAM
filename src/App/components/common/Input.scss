.input {
    font-size: 18px;
    padding: 4px;
    outline: none;
    color: #fefefe;
    background-color: lighten($color: #E54B4B, $amount: 5);
    transition: 0.2s;
}
.input:focus {
    outline: 1px solid #EDAE49;
    background-color: lighten($color: #E54B4B, $amount: 10);
}