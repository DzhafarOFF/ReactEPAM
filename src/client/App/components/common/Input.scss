.input {
    font-size: 18px;
    padding: 4px;
    outline: none;
    border: none;
    border-bottom: 1px solid #fefefe ;
    color: #fefefe;
    background-color: #131313;
    transition: 0.2s;
}
.input:focus {
    border-bottom: 1px solid #EDAE49;
    background-color: lighten($color: #E54B4B, $amount: 10);
}