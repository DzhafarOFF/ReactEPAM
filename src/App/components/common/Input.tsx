import React from 'react';
import './Input.scss';

type InputOptions = {
    type: string;
    changeAction: (e: React.ChangeEvent) => void;
    submitAction: (e: React.KeyboardEvent) => void;
    placeholder?: string;
}

const Input = (props: InputOptions) => {
    return (
        <input
            className = 'input'
            type = {props.type}
            onKeyUp = {props.submitAction}
            onChange = {props.changeAction}
        />
    )
}

export default Input;