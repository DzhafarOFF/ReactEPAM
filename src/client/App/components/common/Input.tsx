import './Input.scss';
import React from 'react';

type Props = {
	type: string
	changeAction?: (e: React.ChangeEvent) => void
	submitAction: (e: React.KeyboardEvent) => void
	placeholder?: string
};

const Input: React.FC<Props> = (props: Props) =>
	<input
		className = 'input'
		type = {props.type}
		onKeyUp = {props.submitAction}
		onChange = {props.changeAction}
	/>
;


export default Input;
