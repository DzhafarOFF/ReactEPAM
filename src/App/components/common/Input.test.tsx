import Input from './Input';
import React from 'react';
import { shallow, render } from 'enzyme';
const changeAction = jest.fn();
const submitAction = jest.fn();
const props = {
    type: 'text',
    changeAction: changeAction,
    submitAction: submitAction,
}
const input = shallow(<Input {...props} />);

describe('<Input> rendering', () => {
    it('should renders correctly', () => {
        expect(input).toMatchSnapshot()
    })

    it('should has correct className', () => {
        expect(input.find('input').hasClass('input')).toEqual(true);
    })

    it('should has correct type', () => {
        expect(input.find('input').props().type).toEqual('text');
    }) 
});

describe('<Input> interaction', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    
    it('should handle onChange event', () => {
        input.find('input').simulate('change',
            { 
                target: {
                    value: '1'
                }
            }
        ).simulate('change',
            { 
                target: {
                    value: '12'
                }
            }
        );
        expect(changeAction.mock.calls.length).toEqual(2);
    });

    it('should handle onKeyUp event', () => {
        input.find('input').simulate('keyup', {key: '1'});
        expect(submitAction.mock.calls.length).toEqual(1);
    });
});