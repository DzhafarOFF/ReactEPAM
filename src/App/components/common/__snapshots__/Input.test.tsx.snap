// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> rendering should renders correctly 1`] = `
<input
  className="input"
  onChange={[MockFunction]}
  onKeyUp={[MockFunction]}
  type="text"
/>
`;
